data_tools/buildhub/utils.py

```diff
@@ -0,0 +1,93 @@
+from os.path import basename
+
+
+def build_record_from_url(url):
+    record = {}
+    parts = basename(url).split('.')
+    url_parts = url.split('/')
+
+    product_name, major_version = parts[0].split('-')
```

Comment:
The products tuple doesn't contains fennec, so I cannot use it to make my test to pass.

We decided together that the ID should be generated from the URL and the URL only.
Then you asked that it comes from the record, but the way the record is currently generated doesn't provide the information to build the ID from it.

You want me to iterate but in the meantime you are the one asking for those refactoring because without them the tests don't pass.

I would be glad to pair review this so that we can discuss our views on the project and find a consensus.